src/sort/simple-sort-directive.js

```diff
@@ -0,0 +1,239 @@
+            count: 2,
+            description: 'Get it while it lasts'
+          },
+          {
+            name: "Item 5",
+            count: 321,
+            description: 'Beautiful style'
+          }
+        ];
+
+        var compareFn = function(item1, item2) {
+          var compValue = 0;
+          if ($scope.sortConfig.currentField === 'name') {
+            compValue = item1.name.localeCompare(item2.name);
+          } else if ($scope.sortConfig.currentField === 'count') {
+              compValue = item1.count - item2.count;
+          } else if ($scope.sortConfig.currentField === 'description') {
+            compValue = item1.description.localeCompare(item2.description);
+          }
+
+          if ($scope.sortConfig.currentDirection === 'desc') {
+            compValue = compValue * -1;
+          }
+
+          return compValue;
+        };
+
+        var sortChange = function (sortId, sortDirection) {
+          $scope.items.sort(compareFn);
+        };
+
+        $scope.sortConfig = {
+          fields: [
+            {
+              id: 'name',
+              title:  'Name',
+              sortType: 'alpha'
+            },
+            {
+              id: 'count',
+              title:  'Count',
+              sortType: 'numeric'
+            },
+            {
+              id: 'description',
+              title:  'Description',
+              sortType: 'alpha'
+            }
+          ],
+          onSortChange: sortChange
+        };
+      }
+    ]);
+  </file>
+</example>
+ */
+angular.module('patternfly.sort').directive('pfSimpleSort',
+  function ($document) {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        config: '='
+      },
+      transclude: false,
+      templateUrl: 'sort/simple-sort.html',
+      controller: function ($scope) {
+        $scope.setupConfig = function () {
+          var updated = false;
+
+          if ($scope.config.fields === undefined) {
+            $scope.config.fields = [];
+          }
+
+          if ($scope.config.fields.length > 0) {
+            if ($scope.config.currentField === undefined) {
+              $scope.config.currentField = $scope.config.fields[0].id;
+              updated = true;
+            }
+            if ($scope.config.currentDirection === undefined) {
+              $scope.config.currentDirection = 'asc';
+              updated = true;
+            }
+          }
+
+          $scope.config.fields.forEach(function (field) {
+            if (field.id === $scope.config.currentField) {
+              $scope.config.currentSortField = field;
+            }
+          });
+
+          if (updated === true && $scope.config.onSortChange) {
+            $scope.config.onSortChange($scope.config.currentField, $scope.config.currentDirection);
+          }
+        };
+
+        $scope.selectField = function (field) {
+          $scope.config.currentField = field.id;
+          $scope.config.currentSortField = field;
```

Comment:
Why not just set `$scope.config.currentField` and then just reference the ID from there (ala `$scope.config.currentField.id`) when you need it?